packages/blocks/src/page-block/edgeless/toolbar/edgeless-toolbar.ts

```diff
@@ -210,10 +206,41 @@ export class EdgelessToolbar extends LitElement {
       y = centerY - (options.height * zoom) / 2;
     }
 
-    this.edgeless.addNewFrame(models, new Point(x, y), options);
+    const { frameId } = this.edgeless.addNewFrame(
+      models,
+      new Point(x, y),
+      options
+    );
+    const frame = this.edgeless.frames.find(frame => frame.id === frameId);
+    assertExists(frame);
+    this.edgeless.slots.mouseModeUpdated.emit({ type: 'default' });
+    this.edgeless.slots.selectionUpdated.emit({
+      selected: [frame],
+      active: false,
+    });
+
     this._imageLoading = false;
   }
 
+  setMouseMode = (mouseMode: MouseMode) => {
+    if (this.edgeless.mouseMode === mouseMode) return;
+    const state = {
+      selected: [],
+      active: false,
+    };
+    if (mouseMode.type === 'default') {
+      if (this.edgeless.selection.lastState) {
+        Object.assign(state, this.edgeless.selection.lastState);
+        this.edgeless.selection.lastState = null;
+      }
+    } else if (this.edgeless.selection.state.selected.length) {
+      this.edgeless.selection.lastState = this.edgeless.selection.state;
+    }
+
+    this.edgeless.slots.mouseModeUpdated.emit(mouseMode);
+    this.edgeless.slots.selectionUpdated.emit(state);
+  };
```

Comment:
It would be better if you designed an API in the edgeless selection manager instead of operating the state directly.